frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-root-page/segment-root-page-content/segment-root-section-card/segment-root-section-card.component.ts

```diff
@@ -0,0 +1,169 @@
+import { RouterModule } from '@angular/router';
+import { MatTableDataSource } from '@angular/material/table';
+import { DialogService } from '../../../../../../../shared/services/common-dialog.service';
+import { Observable, map } from 'rxjs';
+import { Segment } from '../../../../../../../core/segments/store/segments.model';
+import { CommonSearchWidgetSearchParams } from '../../../../../../../shared-standalone-component-lib/components/common-section-card-search-header/common-section-card-search-header.component';
+import {
+  CommonTableHelpersService,
+  TableState,
+} from '../../../../../../../shared/services/common-table-helpers.service';
+import { UserPermission } from '../../../../../../../core/auth/store/auth.models';
+import { AuthService } from '../../../../../../../core/auth/auth.service';
+
+@Component({
+  selector: 'app-segment-root-section-card',
+  imports: [
+    CommonSectionCardComponent,
+    CommonSectionCardSearchHeaderComponent,
+    CommonSectionCardActionButtonsComponent,
+    SegmentRootSectionCardTableComponent,
+    AsyncPipe,
+    NgIf,
+    MatProgressSpinnerModule,
+    RouterModule,
+    TranslateModule,
+    TitleCasePipe,
+  ],
+  templateUrl: './segment-root-section-card.component.html',
+  styleUrl: './segment-root-section-card.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SegmentRootSectionCardComponent {
+  permissions$: Observable<UserPermission>;
+  dataSource$: Observable<MatTableDataSource<Segment>>;
+  isLoadingSegments$ = this.segmentsService.isLoadingSegments$;
+  isInitialLoading$ = this.segmentsService.isInitialSegmentsLoading();
+  isAllSegmentsFetched$; // TODO: Implement if needed for pagination
+  searchString$ = this.segmentsService.selectSearchString$;
+  searchKey$ = this.segmentsService.selectSearchKey$;
+  searchParams$ = this.segmentsService.selectSearchSegmentParams();
+  selectRootTableState$; // TODO: Implement if needed
+  isSearchActive$: Observable<boolean> = this.searchString$.pipe(map((searchString) => !!searchString));
+
+  segmentFilterOptions = [
+    SEGMENT_SEARCH_KEY.ALL,
+    SEGMENT_SEARCH_KEY.NAME,
+    SEGMENT_SEARCH_KEY.STATUS,
+    SEGMENT_SEARCH_KEY.CONTEXT,
+    SEGMENT_SEARCH_KEY.TAG,
+  ];
+  isSectionCardExpanded = true;
+
+  menuButtonItems: IMenuButtonItem[] = [
+    {
+      name: this.translateService.instant('segments.import-segment.text'),
+      disabled: false,
+    },
+    {
+      name: this.translateService.instant('segments.export-all-segments.text'),
+      disabled: true,
+    },
+  ];
+
+  constructor(
+    private segmentsService: SegmentsService,
+    private translateService: TranslateService,
+    private dialogService: DialogService,
+    private tableHelpersService: CommonTableHelpersService,
+    private authService: AuthService
+  ) {}
+
+  ngOnInit() {
+    this.permissions$ = this.authService.userPermissions$;
+    this.segmentsService.fetchSegments(true);
+  }
+
+  ngAfterViewInit() {
```

Comment:
let's not preemptively include the table data logic or sort / filter logic yet, I don't think this is going to be useful for how we'll end up implementing it for a couple of reasons:

- the data will be fetched differently (2 data sources, global and paginated)
- we should use the common table helper service we developed during FF so we're not duplicating search / filter logic.